lib/api/api_auth.go

```diff
@@ -27,14 +28,38 @@ const (
 	randomTokenLength  = 64
 )
 
-func emitLoginAttempt(success bool, username, address string, evLogger events.Logger) {
-	evLogger.Log(events.LoginAttempt, map[string]interface{}{
+func emitLoginAttempt(success bool, username string, r *http.Request, evLogger events.Logger) {
+	remoteIP := osutil.IPFromString(r.RemoteAddr)
+	remoteAddress := remoteIP.String()
```

Comment:
remoteIP could be nil here?

Reply:
Indeed. Apparently it's `@` for clients connecting via unix socket...